### PR TITLE
Add offsetof functionality

### PR DESCRIPTION
**Pull request by Jopyth**
It does not belong into any of the existing folders, any (naming) suggestions on the category of this kind of functionality?

### CODE REVIEW

**Comment by sbusch42** [comment]
What about 'memory'? Could get other members such as make_unique etc. (if needed)

**Comment by scheibel** [comment]
I like the idea of `memory` although I wonder where to discriminate between functionality like `make_unique` and `contains` (e.g., on strings or containers).
Basically, I wish for another module where STL extensions can be located. `make_unique` would be such an STL extension, `contains` would also be. However, `offsetof` is more or less a language or compiler extension, not relying on the STL at all.

**Comment by scheibel** [comment]
I also want to add the rationale for this functionality:

There is an inofficial offsetof language extension that is mainly implemented as compiler extension but it is rather limited in usage. Lately, some compilers even dropped support of this functionality.
However, I think we understand the limitations of the implementation and won't use it in problematic cases.

**Comment by scheibel** [comment]
This PR is part of #6.

File: source/cppassist/include/cppassist/offsetof.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+#pragma once
+
+#include <cstddef>
+
+namespace cppassist
```
Comment (scheibel) [inline source/cppassist/include/cppassist/offsetof.h]: ANother empty line here, please.

**Pull request merged**